domains/transfer-agreements/TransferAgreements.API/API/Data/TransferAgreementRepository.cs

```diff
@@ -35,4 +35,21 @@ public async Task<List<TransferAgreement>> GetTransferAgreementsList(Guid subjec
             .Where(agreement => agreement.Id == id && (agreement.SenderId == Guid.Parse(subject) || agreement.ReceiverTin.Equals(tin)))
             .FirstOrDefaultAsync();
     }
+
+    public async Task Save()
+    {
+        await context.SaveChangesAsync();
+    }
+
+    public async Task<TransferAgreement?> GetTransferAgreement(Guid id) =>
+        await context.TransferAgreements
+            .FirstOrDefaultAsync(t => t.Id == id);
+
+    public async Task<bool> HasDateOverlap(Guid id, DateTimeOffset endDate, Guid senderId, string receiverTin) =>
```

Comment:
I think this public method can be avoided if we do it like in the contract service